src/Utils.ts

```diff
@@ -986,12 +986,129 @@ function setDummyInputDebugValue(
     );
 }
 
+export class DummyInputObserver implements Types.DummyInputObserver {
+    private _win?: GetWindow;
+    private _offsetsQueue: (() => void)[] = [];
+    private _offsetsTimer?: number;
+    private _domChangedMap: WeakMap<HTMLElement, true> = new WeakMap();
+    private _domChangedTimer?: number;
```

Comment:
```suggestion
    private _pendingDummyInputResets: WeakSet<HTMLElement> = new WeakSet();
    private _resetDummyInputsTimer?: number;
```
I'd rename these to be clearer what they do, also if you're just mapping something to a boolean, you can use a [WeakSet](https://caniuse.com/?search=JavaScript%20WeakSet).

The same comment applies about the weak structure, there is an explicit dispose step so I don't think weak structures are necessary

Reply:
`_pendingChangedParents`?

With these names I also always struggle to make them shorter (as the prop names do not get minified at all).

Reply:
> With these names I also always struggle to make them shorter (as the prop names do not get minified at all).

It's a cost of using ES6 classes (we can stop doing that at anytime), you can do some kind of hungarian notation here, but `domChangedMap` actually means nothing to what is stored in there

I'm ok with `_pendingChangedParents`